javascript/javascript2/week2/homework/Movies_and_oddDoubledNumbers/moviesTasks.js

```diff
@@ -0,0 +1,118 @@
+// just for reference
+console.log(movies.length);
+
+// 1. To create an array of movies containing the movies with a short title
+const shortMovieTitles = movies.filter(movies => movies.title.length < 6);
+console.log(shortMovieTitles);
+
+// 2. to create an array of movie titles with long movie titles
+const longMovieTitles = movies.filter(movies => movies.title.length > 65);
+console.log(longMovieTitles);
+
+// 3. to count the number of movies made between 1980-1989 (including both the years)
+const moviesBetween1980And1989 = movies.filter(movies => movies.year >= 1980 && movies.year <= 1989);
+console.log(moviesBetween1980And1989);
+
+// 4. Create a new array with extra key called 'tag'. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const extraKeyCalledTag = movies.map(movie => {
+	if (movie.rating >= 7) {
+		movie.tag = `Good`;
+		return movie;
+	} else if (movie.rating >= 4) {
+		movie.tag = `Average`;
+		return movie;
+	} else if (movie.rating < 4) {
+		movie.tag = `Bad`;
+		return movie;
+	}
+});
+
+console.log(extraKeyCalledTag);
+
+// 5. chaining, movies rated higher than 6
+
+const moviesRatedHigherThan6 = movies
+	.filter(movie => movie.rating > 6)
+	.map(movie => `movie : ${movie.title}, rating : ${movie.rating}`);
+
+console.log(moviesRatedHigherThan6);
+
+// 6. Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin
+
+const countAllMoviesContainingKeywords = movies.filter(movie => {
+	const movieNamesArray = movie.title.toLowerCase().split(' ');
+	if (
+		movieNamesArray.includes('surfer') ||
+		movieNamesArray.includes('alien') ||
+		movieNamesArray.includes('benjamin')
+	) {
+		return movie;
+	}
+}).length;
+console.log(countAllMoviesContainingKeywords);
+
+
+// 7. Create an array of movies where a word in the title is duplicated. Fx "Star Wars: The Clone Wars" the word Wars is duplicated.
+const duplicatedWordsInTitles = [];
+
+movies.map(movie => movie.title.split(' '))
+	.forEach(title => title.filter((word, index) => title.indexOf(word) !== index ? duplicatedWordsInTitles.push(word) : false));
+console.log(duplicatedWordsInTitles);
+
+// 8. Find the word that is mostly duplicated using sort
+
+let wordCount = [];
+
+movies
```

Comment:
this is really really slow to run, the reason is that finding things in arrays is really slow in computers, when we have to repeatedly lookup a value by a name, we would use an object [also called a hash in other languages]

each time your loop runs we need to calculate pos, we need to calculate pos again, and search the entire list to find the word.

if you stored your data in an object
```
{ "the": 2423, "a": 2421 ....
```
then it would be much quicker to lookup the words because we could find them using wordCount['the'] for example you can convert an object to an array for sorting using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

what i'm describing here is similar to how you calculate `countTotalNumber` further down in this file